File: src/main/java/ru/rsreu/astrukov/bool/BoolApplication.java
package ru.rsreu.astrukov.bool;

import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;
import ru.rsreu.astrukov.bool.model.BoolElement;
import ru.rsreu.astrukov.bool.service.EquationSolverKt;

import javax.swing.text.html.HTMLDocument;
import java.util.Arrays;


public class BoolApplication extends Application {

    @Override
    public void start(Stage stage) throws Exception {

        Parent sceneWithLayout = FXMLLoader.load(getClass().getResource("/scene.fxml"));


        Scene scene = new Scene(sceneWithLayout, 800, 600);
        stage.setTitle("BoolApplication");
        stage.setScene(scene);
        stage.show();

        EquationSolverKt eqs = new EquationSolverKt();

//        eqs.tst();

        String fn = "(x1 && !x2 && !x3 && x5) ||" +
                " (!x1 && x3 && !x5) || (x1 && x3 && !x4 && !x5)" +
                " || (x1 && x2 && !x3 && x5)";

        BoolElement root = eqs.solve(
                fn
                ,
                Arrays.asList("x1", "x2", "x3", "x4", "x5"));

        String a = "";
    }


    public static void main(String[] args) {
        launch(args);
    }

}

File: src/main/java/ru/rsreu/astrukov/bool/BoolApplication.kt
package ru.rsreu.astrukov.bool

import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.stage.Stage
import ru.rsreu.astrukov.bool.model.*
import ru.rsreu.astrukov.bool.model.element.BoolElement
import ru.rsreu.astrukov.bool.model.element.BoolElementBlock
import ru.rsreu.astrukov.bool.service.EquationSolver


class BoolApplication : Application() {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            launch(BoolApplication::class.java)
        }
    }

    override fun start(stage: Stage) {

        val sceneWithLayout = FXMLLoader.load<Parent>(javaClass.getResource("/scene.fxml"))

        val scene = Scene(sceneWithLayout, 800.0, 600.0)
        stage.title = "BoolApplication"
        stage.scene = scene
        stage.show()

    }

}
